src/cse-machine/interpreter.ts

```diff
@@ -98,15 +98,34 @@ type CmdEvaluator = (
  * It contains syntax tree nodes or instructions.
  */
 export class Control extends Stack<ControlItem> {
+  private numEnvDependentItems: number
   public constructor(program?: es.Program | StatementSequence) {
     super()
-
+    this.numEnvDependentItems = 0
     // Load program into control stack
     program ? this.push(program) : null
   }
 
+  public canAvoidEnvInstr(): boolean {
+    return this.numEnvDependentItems == 0
```

Comment:
Let's use strict equality here

```suggestion
    return this.numEnvDependentItems === 0
```

Reply:
Resolved in eca7bbf